sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -130,10 +130,12 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case CreateTable(tableDesc, mode, None) if DDLUtils.isDatasourceTable(tableDesc) =>
+      DDLUtils.checkFieldNames(tableDesc)
       CreateDataSourceTableCommand(tableDesc, ignoreIfExists = mode == SaveMode.Ignore)
 
     case CreateTable(tableDesc, mode, Some(query))
         if query.resolved && DDLUtils.isDatasourceTable(tableDesc) =>
+      DDLUtils.checkFieldNames(tableDesc.copy(schema = query.schema))
       CreateDataSourceTableAsSelectCommand(tableDesc, mode, query)
 
     case InsertIntoTable(l @ LogicalRelation(_: InsertableRelation, _, _, _),
```

Comment:
Should we block `InsertIntoTable(l @ LogicalRelation(t: HadoopFsRelation...` too?

Reply:
Ur, if you don't mind, I'll proceed `INSERT INTO` as another PR.

Reply:
Since this PR is not big, it should cover all these scenarios.

Reply:
Sorry, but I'm not sure when `INSERT INTO TABLE` has this kind of issue.
In case of `INSERT INTO`, the table already exists.

Reply:
So far, it looks different than CTAS.

Reply:
You did the check for Hive serde tables, but no check is done in data source tables?

Reply:
Oh, I'll remove it from Hive serde table case. Checking on the existing table during INSERT INTO seems to be actually no-op.